### ISSUE TITLE
Suggestion for added resiliency for consumers without address resolver

### ISSUE THREAD

**Issue by ricsiLT** [rabbitmq/rabbitmq-stream-dotnet-client#123]
Hi,

We have an issue with our network which prompted me to look at `RoutingClient` again and I wanted to run my suggestion past you before creating a PR for this.

Our setup is:

* no address resolver
* DnsEndpoints provided
* We provide 4 endpoints even though metadata will report 5 (5th node is unreachable from client-side and is present to break stalemates in voting I guess)

In such case, randomly connecting as a consumer might select unreachable endpoint and crash with SocketException.

In the minimum case, I'd like to change current impl of `LookupRandomConnection` with (comments removed for brevity):

```
        public static async Task<IClient> LookupRandomConnection(ClientParameters clientParameters,
            StreamInfo metaDataInfo)
        {
            var brokers = new List<Broker>() { metaDataInfo.Leader };
            brokers.AddRange(metaDataInfo.Replicas);
            var rnd = new Random();
            brokers.Sort((_, _) => rnd.Next(-1, 1));
            foreach (var broker in brokers)
            {
                try
                {
                    return await LookupConnection(clientParameters, broker, MaxAttempts(metaDataInfo));
                }
                catch (SocketException)
                {
                }
            }
            throw new RoutingClientException("None of the brokers could be reached");
        }
```

This could also be handled in `LookupConnection` I suppose. Still, I'm not sure whether this needs to be handled in library at all, that's why I wanted to run this by you guys.

As a bonus, would you be ok with wrapping SocketExceptions with, lets say, `ClientConnectionException` which would contain ip/port that library attempted to reach?

**Comment by lukebakken** [comment]
It seems like a good idea to try the full set of brokers when connecting. I would like to see the exception logged at the `debug` level.

> As a bonus, would you be ok with wrapping SocketExceptions with, lets say, ClientConnectionException which would contain ip/port that library attempted to reach?

Hm, we don't do that in the AMQP .NET client.

**Comment by ricardSiliuk** [comment]
PR here: https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/124

**Comment by lukebakken** [comment]
Fixed by #124
